### PR TITLE
[#501] Add FDA Search UI page

### PR DESCRIPTION
**Pull request by vitorbaptista**
opentrials/opentrials#501

### CODE REVIEW

**Comment by nightsh** [comment]
@vitorbaptista I can have a look at it now

**Comment by vitorbaptista** [comment]
@nightsh This is WIP and very early stage, so no need to review now :+1:

**Comment by nightsh** [comment]
I was just curious :smile: 
There are a quite few interesting `hapi` things in there already, I'm learning new things every day!

I'll have a more thorough look after you remove the WIP tag, until then I'll just be happy to be notified when things happen (it's quite an exciting PR, to be honest).

**Comment by vitorbaptista** [comment]
@nightsh this is ready for review

**Comment by nightsh** [comment]
Testing it before reading, spotted this on searching by date:

```
get /search {"q":"","condition":["Tick-borne Encephalitis","Comparison of Energy Intake Methods"],"intervention":["Vitamin D3 oil three times in a year","fentanyl/bupivacaine/epinephrine"],"person":["Clemens von.Birgelen, MD,Phd","Amelie G Ramirez, DRPH","Kristen Rosen, MPH"],"organisation":["Elan Pharma Ltd","ALTUS PHARMACEUTICALS INC"],"location":["Malasya","India"],"sample_size_start":"10","sample_size_end":"20","gender":"female","has_published_results":"true","has_publications":"false","has_discrepancies":"true","registration_date_start":"11/09/2016","registration_date_end":"08/27/2020","advanced_search":false} 400 (25ms)
```

I'll go through the code now, then I'll probably attempt some more clicking around in case I missed something. I'll let you know.

File: views/fda/partials/search-form-fields.html [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    value="{{ value }}"
+    {% if selected %}selected{% endif %}
+    >
+    {{ value }}
+  </option>
+{% endmacro %}
+
+<fieldset class="search-bar">
+  <label>Drug, ingredients, or other keywords</label>
+  <input type="search" title="Search" placeholder="e.g. &quot;heart attack&quot;" name="q" value="{{ query.q }}" autofocus />
```
Comment (nightsh) [inline views/fda/partials/search-form-fields.html]: Not sure it matters, but the example value returns no hits and a suggestion to also try searching in the documents text. Might want to change it in something that returns results, like "Aspirin" or smth else.

File: views/fda/partials/search-form-fields.html [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    </div>
+  </div>
+
+  <fieldset class="registration date">
+      <legend style="display:none;">Action period</legend>
+      <div class="filter">
+        <label for="action_date_start">Start date</label>
+        <input
+          type="date"
+          data-format="dd/MMM/yyyy"
```
Comment (nightsh) [inline views/fda/partials/search-form-fields.html]: This! Will fix opentrials/opentrials#565

File: views/fda/partials/search-form-fields.html (hunk 1 above)
Comment (vitorbaptista) [inline views/fda/partials/search-form-fields.html]: Good point! Changed to aspirin.

**Pull request merged**